typed-racket-lib/typed-racket/typecheck/tc-toplevel.rkt

```diff
@@ -479,6 +479,7 @@
       (define/with-syntax (new-defs ...) defs)
       (define/with-syntax (new-export-defs ...) export-defs)
       (define/with-syntax (new-provs ...) provs)
+      (do-contract-requires)
       (values
```

Comment:
is it important that `do-contract-requires` appears here (and not earlier or later)?  Does it just need to come before the `get-contract-requires` below? If so, would it be clearer to have a function that just does both in the right order? (i.e. does the work of `do-contract-requires` and then returns the result from `get-contract-requires`?)

Reply:
Or if not, maybe a comment about why this effectful procedure is called there?

Reply:
No, `do-contract-requires` is for side-effect before any contracts are generated, so it has to be earlier. I'll find the logical place to put it an add a comment. `get-contract-requires` doesn't depend on those side effects, and just needs to be called to put the output in the correct submodule.